worlds/aquaria/Regions.py

```diff
@@ -0,0 +1,1377 @@
+"""
+Author: Louis M
+Date: Fri, 15 Mar 2024 18:41:40 +0000
+Description: Used to manage Regions in the Aquaria game multiworld randomizer
+"""
+
+from typing import Dict, Optional
+from BaseClasses import MultiWorld, Region, Entrance, ItemClassification, LocationProgressType
+from .Items import AquariaItem
+from .Locations import AquariaLocations, AquariaLocation
+from .Options import AquariaOptions
+from worlds.generic.Rules import add_rule, set_rule
+
+
+# Every condition to connect regions
+
+def _has_hot_soup(state, player: int) -> bool:
+    """`player` in `state` has the hotsoup item"""
+    return state.has("Hot soup", player)
+
+
+def _has_tongue_cleared(state, player: int) -> bool:
+    """`player` in `state` has the Body tongue cleared item"""
+    return state.has("Body tongue cleared", player)
+
+def _has_sun_crystal(state, player: int) -> bool:
+    """`player` in `state` has the Sun crystal item"""
+    return state.has("Has sun crystal", player)
+
+
+def _has_li(state, player: int) -> bool:
+    """`player` in `state` has Li in it's team"""
+    return state.has("Li and Li song", player)
+
+
+def _has_damaging_item(state, player: int) -> bool:
+    """`player` in `state` has the shield song item"""
+    return (state.has("Energy form", player) or
+            state.has("Nature form", player) or
+            state.has("Beast form", player) or
+            state.has("Li and Li song", player) or
+            state.has("Baby nautilus", player) or
+            state.has("Baby piranha", player) or
+            state.has("Baby blaster", player)
+            )
```

Comment:
```suggestion
def _has_damaging_item(state, player: int) -> bool:
    """`player` in `state` has the shield song item"""
    return state.has_any({"Energy form", "Nature form", "Beast form", "Li and Li song", "Baby nautilus", "Baby piranha", "Baby blaster"}, player)
```

You can use `state.has_any` to check if the player has any one of the items in a set